compiler/vm/vmgen.nim

```diff
@@ -1482,18 +1528,11 @@ proc setSlot(c: var TCtx; v: PSym) =
   if v.position == 0:
     v.position = getFreeRegister(c, if v.kind == skLet: slotFixedLet else: slotFixedVar, start = 1)
 
-proc cannotEval(c: TCtx; n: PNode) {.noinline.} =
-  globalReport(c.config, n.info, reportAst(rsemVmCannotEvaluateAtComptime, n))
-  # HACK REFACTOR FIXME With current compiler 'arhitecture' this call
-  # MUST raise an exception that is captured by `sem.tryConstExpr` in sem. In
-  # the future this needs to be removed, `checkCanEval` must return a
-  # `true/false` bool.
-  #
-  # For more elaborate explanation of the related code see the comment
-  # https://github.com/nim-works/nimskull/pull/94#issuecomment-1006927599
-  #
-  # This code must not be reached
-  raiseRecoverableError("vmgen.cannotEval failed")
+func cannotEval(c: TCtx; n: PNode) {.noinline, noreturn.} =
```

Comment:
This isn't a hack as in we shouldn't avoid the raise?

Reply:
I meant the exception based signaling, but it works for now. Thanks for providing your perspective.